### PR TITLE
DD4hep: Subtraction Solid

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Add check if current solid is a subtraction
* Add a helper to retrieve solidA (aka left shape)
* Add a unit test

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@slomeo - FYI ```isASubtraction()``` is implemented here.

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 1]
```diff
@@ -13,6 +13,19 @@ using namespace edm;
 using namespace std;
 using namespace cms::dd;
 
+dd4hep::Solid DDSolid::solidA() const {
+  if (dd4hep::instanceOf<dd4hep::SubtractionSolid>(solid_) || dd4hep::instanceOf<dd4hep::UnionSolid>(solid_) ||
+      dd4hep::instanceOf<dd4hep::IntersectionSolid>(solid_)) {
+    const TGeoCompositeShape* sh = (const TGeoCompositeShape*)solid_.ptr();
+    const TGeoBoolNode* boolean = sh->GetBoolNode();
+    TGeoShape* solidA = boolean->GetLeftShape();
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDFilteredView.cc]: Seems a bit confusing to re-use the name of the member function within that function. How about instead:
```
TGeoShape* solidLeft = boolean->GetLeftShape();
```

Comment (fabiocos) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: I would naively agree with @cvuosalo comment, although the original code is strictly speaking not wrong

Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: I think it's a question of a style. I find it more clear that the method that requests ```solidA()``` declares that the ```solidA``` is a left shape of a boolean.

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
@smuzaffar is the bot stuck?

**Comment by smuzaffar** [comment]
@fabiocos , might be issue with github web hooks

**Pull request merged**